### PR TITLE
Improve positioning and size of the OpenGL window

### PR DESCRIPTION
**Pull request by hydrobeam**
## Motivation

Before, the opengl window would be set based on the resolution of the scene and wouldn't move to a specific position, which was a very bad solution since the window would very often go beyond the edge of the screen making it effectively unusable. This should be a big improvement to the usability of the opengl renderer. Most of the solution was ported from manimgl, although it was made more generalizable and localized to a single file.

## Explanation for Changes

Before:
image

New:
image

### Added new flags

- `fullscreen`: based on the `frame_width`/ `frame_height` (allows for varying screen sizes), expands the renderer to take up the entirety of the screen. `False` by default
   - (not actually fullscreen because my screen ratio is 3:2 vs manim's 16:9) 
   - image

- `window_position`: by default `UR`. Takes in `L, R, D, U, O (origin)` and moves the window to that position. Also accepts pixel position of the top left corner of the window. 
- `window_monitor`: by default `0`. The index of the monitor to be used.   

### Testing code
```py
class Testing(Scene):
    def construct(self):
        a = Circle()
        self.play(Write(a))
        self.interactive_embed()
```

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by TRoboto** [comment]
Thank you for your contribution. I have tried using these changes but had few issues.

1- Specifying `fullscreen = True` from the `cfg` file doesn't work.
2- `window_position` only works with `UL, UR, DL, DR` and produces the following error when using `L, R, D, U, O`.
3- `window_position` and `window_monitor` cannot be specified from the cli. It outputs `Error: No such option`

<details><summary>Window position error message</summary>

```py
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /media/roboto/work/Repos/manim/manim/cli/render/commands.py:108 in render                        │
│                                                                                                  │
│   105 │   │   │   │   while True:                                                                │
│   106 │   │   │   │   │   scene_classes = scene_classes_from_file(file)                          │
│   107 │   │   │   │   │   SceneClass = scene_classes[0]                                          │
│ ❱ 108 │   │   │   │   │   scene = SceneClass(renderer)                                           │
│   109 │   │   │   │   │   status = scene.render()                                                │
│   110 │   │   │   │   │   if status:                                                             │
│   111 │   │   │   │   │   │   continue                                                           │
│                                                                                                  │
│ /media/roboto/work/Repos/manim/manim/scene/scene.py:128 in __init__                              │
│                                                                                                  │
│    125 │   │   │   )                                                                             │
│    126 │   │   else:                                                                             │
│    127 │   │   │   self.renderer = renderer                                                      │
│ ❱  128 │   │   self.renderer.init_scene(self)                                                    │
│    129 │   │                                                                                     │
│    130 │   │   self.mobjects = []                                                                │
│    131 │   │   # TODO, remove need for foreground mobjects                                       │
│                                                                                                  │
│ /media/roboto/work/Repos/manim/manim/renderer/opengl_renderer.py:225 in init_scene               │
│                                                                                                  │
│   222 │   │   self.scene = scene                                                                 │
│   223 │   │   if not hasattr(self, "window"):                                                    │
│   224 │   │   │   if config["preview"]:                                                          │
│ ❱ 225 │   │   │   │   self.window = Window(self)                                                 │
│   226 │   │   │   │   self.context = self.window.ctx                                             │
│   227 │   │   │   │   self.frame_buffer_object = self.context.detect_framebuffer()               │
│   228 │   │   │   else:                                                                          │
│                                                                                                  │
│ /media/roboto/work/Repos/manim/manim/renderer/opengl_renderer_window.py:46 in __init__           │
│                                                                                                  │
│   43 │   │                                                                                       │
│   44 │   │   self.swap_buffers()                                                                 │
│   45 │   │                                                                                       │
│ ❱ 46 │   │   initial_position = self.find_initial_position(size, monitor)                        │
│   47 │   │   self.position = initial_position                                                    │
│   48 │                                                                                           │
│   49 │   # Delegate event handling to scene.                                                     │
│                                                                                                  │
│ /media/roboto/work/Repos/manim/manim/renderer/opengl_renderer_window.py:96 in                    │
│ find_initial_position                                                                            │
│                                                                                                  │
│   93 │   │   height_diff = monitor.height - window_height                                        │
│   94 │   │                                                                                       │
│   95 │   │   return (                                                                            │
│ ❱ 96 │   │   │   monitor.x + char_to_n[custom_position[1]] * width_diff // 2,                    │
│   97 │   │   │   -monitor.y + char_to_n[custom_position[0]] * height_diff // 2,                  │
│   98 │   │   )                                                                                   │
│   99                                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
IndexError: string index out of range
```
</details>

**Comment by hydrobeam** [comment]
>1- Specifying fullscreen = True from the cfg file doesn't work.
After some testing, this seems to be an issue that applies to all flags that also accept CLI input (-p, -f). I'm not sure what the issue is, but it's probably outside the scope of this PR. 

> 2- window_position only works with UL, UR, DL, DR and produces the following error when using L, R, D, U, O.
I've added an error message that explains that you must specify both Y and X directions (UO, OR). I've also added some checks that would allow you to directly specify ORIGIN/LEFT/DOWN opposed to OO, OL, DO

> 3- window_position and window_monitor cannot be specified from the cli. It outputs Error: No such option
It would be possible to make it so that these can be called via the cli, but given the issue with 1. It's probably more convenient to set those in the config file as opposed to the command line since it's something you'd only have to configure once to find 
 your preference.

**Comment by TRoboto** [comment]
> After some testing, this seems to be an issue that applies to all flags that also accept CLI input (-p, -f). I'm not sure what the issue is, but it's probably outside the scope of this PR.

Hmmm, true. I am also experiencing this issue with other flags.

> It would be possible to make it so that these can be called via the cli, but given the issue with 1. It's probably more convenient to set those in the config file as opposed to the command line since it's something you'd only have to configure once to find your preference.

I agree, it's more convenient to set them in the config file.

File: manim/renderer/opengl_renderer_window.py [hunk 1]
```diff
@@ -28,6 +43,12 @@ def __init__(self, renderer, size=None, **kwargs):
 
         self.swap_buffers()
 
+        initial_position = self.find_initial_position(size, monitor)
+        print(initial_position)
+        print(size)
+        self.position = initial_position
```
Comment (TRoboto) [inline manim/renderer/opengl_renderer_window.py:49]: Did you forget to remove these?

File: manim/renderer/opengl_renderer_window.py [hunk 2]
```diff
@@ -59,3 +80,33 @@ def on_mouse_drag(self, x, y, dx, dy, buttons, modifiers):
...
         self.renderer.scene.on_mouse_drag(point, d_point, buttons, modifiers)
+
+    def find_initial_position(self, size, monitor):
+        custom_position = config.window_position
+        window_width, window_height = size
+        # Position might be specified with a string of the form
+        # x,y for integers x and y
+        if len(custom_position) == 1:
+            raise ValueError(
+                "window_position must specify both Y and X positions (Y/X -> UR). Also accepts LEFT/ORIGIN/UP."
```
Comment (TRoboto) [inline manim/renderer/opengl_renderer_window.py:91]: Maybe add `RIGHT` also at the end?

**Review by TRoboto — APPROVED** [review]
LGTM in general, I left few comments.

File: manim/cli/render/global_options.py [hunk 3]
```diff
@@ -62,4 +62,9 @@ def validate_gui_location(ctx, param, value):
         callback=validate_gui_location,
         help="Starting location for the GUI.",
     ),
+    option(
+        "--fullscreen",
+        is_flag=True,
+        help="Enable GUI interaction.",
```
Comment (TRoboto) [inline manim/cli/render/global_options.py:68]: You forget to change help text

File: manim/renderer/opengl_renderer_window.py (hunk 1 above)
Comment (hydrobeam) [reply manim/renderer/opengl_renderer_window.py]: Whoops 😅. The second `self.position = initial_position` is actually a bug fix from manimgl where it wouldn't set the position sometimes with only one, but I haven't encountered it 🤔

Comment (TRoboto) [reply manim/renderer/opengl_renderer_window.py]: It works fine for me now. If anyone encountered an issue with the initial position, we can find more convenient solution.

**Review by behackl — APPROVED** [review]
I've played around with these changes locally, all works well. Thanks for your efforts!

**Comment by eulertour** [comment]
Is there any way to configure the preview window's size with this?

**Pull request merged**